components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2TokenEventPayloadBuilder.java

```diff
@@ -0,0 +1,143 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.webhook.wso2.event.handler.api.builder;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.core.context.IdentityContext;
+import org.wso2.carbon.identity.core.context.model.Flow;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.publisher.api.model.EventPayload;
+import org.wso2.identity.webhook.common.event.handler.api.builder.TokenEventPayloadBuilder;
+import org.wso2.identity.webhook.common.event.handler.api.constants.Constants;
+import org.wso2.identity.webhook.common.event.handler.api.model.EventData;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.WSO2TokenIssuedEventPayload;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.WSO2TokenRevokedEventPayload;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.common.AccessToken;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.common.Application;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.common.Organization;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.common.Tenant;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.common.User;
+import org.wso2.identity.webhook.wso2.event.handler.internal.model.common.UserStore;
+import org.wso2.identity.webhook.wso2.event.handler.internal.util.WSO2PayloadUtils;
+
+import java.util.Map;
+
+/**
+ * Builder class for creating WSO2 Token Event Payloads.
+ */
+public class WSO2TokenEventPayloadBuilder implements TokenEventPayloadBuilder {
+
+    @Override
+    public Constants.EventSchema getEventSchemaType() {
+
+        return Constants.EventSchema.WSO2;
+    }
+
+    @Override
+    public EventPayload buildAccessTokenRevokeEvent(EventData eventData) throws IdentityEventException {
+
+        //TODO: Implement the logic to build the WSO2TokenRevokedEventPayload from eventData.
+
+        return new WSO2TokenRevokedEventPayload.Builder()
+                .accessTokens(null)
+                .initiatorType(null)
+                .tenant(null)
+                .userStore(null)
+                .user(null)
+                .application(null)
+                .build();
+    }
+
+    @Override
+    public EventPayload buildAccessTokenIssueEvent(EventData eventData) throws IdentityEventException {
+
+        Tenant tenant = WSO2PayloadUtils.buildTenant();
+        UserStore userStore = WSO2PayloadUtils.buildUserStore(eventData);
+        Application application = buildApplication(eventData);
+        AccessToken accessToken = buildAccessToken(eventData);
+        User user = WSO2PayloadUtils.buildUser(eventData);
```

Comment:
need to add the user resident organization to the user. 
Check for all the places

Reply:
df79cc2bff922656d678c24c030cd9e29bcf6327